ocaml/xapi/xapi_host_helpers.ml

```diff
@@ -362,7 +362,23 @@ let consider_enabling_host_nolock ~__context =
          pbds
       )
   in
-  if
+  let host_pending_mandatory_guidances =
```

Comment:
Wouldn't the check be done after the `if !Xapi_globs.on_system_boot` as in the branch, the pending guidance list will be cleared?

Reply:
Oh, the code change here is not correct, thank you!